src/emmodel/model.py

```diff
@@ -52,59 +52,128 @@ def run_models(self):
+            Dataframe that contains information of the model data.
+        col_pred : str, optional
+            Name of the final prediction column, by default "deaths_pred"
+        num_samples : int, optional 
+            Number of samples, by default 1000. Only use when results_samples
+            is None.
+        results_samples : List[List[Dict]], optional
+            Samples of results, by default None. If None will automatically
+            sample results.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            Draws of time_trend, time_residual and col_pred.
+        """
+        if results_samples is None:
+            results_samples = self.get_results_samples(num_samples)
+
+        draws = {"time_trend": [], "time_residual": [], col_pred: []}
+        for results in results_samples:
+            df = self._predict(df.copy(), col_pred=col_pred, results=results)
+            for key in draws.keys():
+                draws[key].append(df[key].to_numpy())
+        for key in draws.keys():
+            draws[key] = np.vstack(draws[key])
+        return draws
+
+    def predict(self,
+                df: pd.DataFrame,
+                col_pred: str = "deaths_pred",
+                include_ci: bool = False,
+                ci_bounds: Tuple[float] = (0.025, 0.975),
+                **kwargs) -> pd.DataFrame:
         """
-        Predict expected deaths ('deaths_pred') from data and model fit with intervals
+        Predict expected deaths('deaths_pred') from data and model fit
         """
-        lst_pred = []
-        lst_time_trend = []
-        lst_trend_residual = []
-        for _ in range(num_draws):
-            df_sim = self.predict(df, col_pred, sim_coef=True)
-            pred = df_sim.loc[:, col_pred].values
-            time_trend = df_sim.loc[:, 'time_trend'].values
-            trend_residual = df_sim.loc[:, 'trend_residual'].values
-            lst_pred.append(pred.tolist())
-            lst_time_trend.append(time_trend.tolist())
-            lst_trend_residual.append(trend_residual.tolist())
-
-        df[f"{col_pred}_mean"], df[f"{col_pred}_lower"], df[f"{col_pred}_upper"] = \
-            summarize_uncertainty(lst_pred)
-        df['trend_residual_mean'], df['trend_residual_lower'], df['trend_residual_upper'] = \
-            summarize_uncertainty(lst_trend_residual)
-        df['time_trend_mean'], df['time_trend_lower'], df['time_trend_upper'] = \
-            summarize_uncertainty(lst_time_trend)
-        df = df.drop(columns=[col_pred, 'trend_residual', 'time_trend'])
+        df = self._predict(df, col_pred=col_pred)
+        if include_ci:
+            draws = self.get_draws(df, col_pred=col_pred, **kwargs)
+            for key in draws.keys():
+                df[f"{key}_lower"] = np.quantile(draws[key], ci_bounds[0], axis=0)
+                df[f"{key}_mean"] = np.mean(draws[key], axis=0)
+                df[f"{key}_upper"] = np.quantile(draws[key], ci_bounds[1], axis=0)
```

Comment:
Do we need to drop columns of `key`? Since both `key` and `key_mean` will exist.